src/Microsoft.Azure.WebJobs.Host/Extensions/Tooling.cs

```diff
@@ -0,0 +1,397 @@
+                    FileAccess access;
+                    switch (token.ToString().ToLowerInvariant())
+                    {
+                        case "in":
+                            access = FileAccess.Read;
+                            break;
+                        case "out":
+                            access = FileAccess.Write;
+                            break;
+                        case "inout":
+                            access = FileAccess.ReadWrite;
+                            break;
+                        default:
+                            throw new InvalidOperationException($"Illegal direction value: '{token}'");
+                    }
+                    metadata["access"] = access.ToString();
+                }
+            }
+
+            // Special case handling of StorageAccountAttribute
+            // $$$ - make StorageAccountAttribute a base class of BlobAttribuet and then this goes aways
+            if (attributeType == typeof(BlobAttribute) ||
+                attributeType == typeof(BlobTriggerAttribute) ||
+                attributeType == typeof(TableAttribute) ||
+                attributeType == typeof(QueueAttribute) ||
+                attributeType == typeof(QueueTriggerAttribute))
+            {
+                if (metadata.TryGetValue("Connection", StringComparison.OrdinalIgnoreCase, out token))
+                {
+                    var attr = new StorageAccountAttribute(token.ToString());
+                    list.Add(attr);
+                }
+            }
+        }
+
+        private bool CanBind(Attribute attribute, Type parameterType)
+        {
+            IBindingProvider root = this._root;
+
+            bool result = Task.Run(() => ScriptHelpers.CanBindAsync(root, attribute, parameterType)).GetAwaiter().GetResult();
+
+            return result;
+        }
+               
+        // Get a better implementation 
+        public Type GetDefaultType(
+            Attribute attribute,
+            FileAccess access, // direction In, Out, In/Out
+            Type requestedType) // combination of Cardinality and DataType            
+        {
+            if (attribute == null)
+            {
+                throw new ArgumentNullException(nameof(attribute));
+            }
+            if (access == FileAccess.Write)
+            {
+                var outType = _outTypes.FirstOrDefault(type => this.CanBind(attribute, type));
+                if (outType != null)
+                {
+                    return outType;
+                }
+                // Error! 
+                throw new InvalidOperationException($"Can't bind {attribute.GetType().Name} to a script-compatible  output.");
+            }
+            else if (access == FileAccess.Read)
+            {
+                if (IsBinary(requestedType))
+                {
+                    return typeof(byte[]);
+                }
+
+                var array = IsCardinalityMany(requestedType);
```

Comment:
renaming to: GetElementTypeIfCardinalityMany


---
In reply to: [108527761](https://github.com/Azure/azure-webjobs-sdk/pull/1093#discussion_r108527761) [](ancestors = 108527761)